### PR TITLE
Fix #1314, shard manager class option cannot be overridden

### LINKED ISSUE #1314: Unable to set custom shard manager
Good Day,

In https://github.com/doctrine/DoctrineBundle/blob/70be9d41da1c75c613d79e9d3c604d179cd705b0/DependencyInjection/DoctrineExtension.php#L405 the `$options['shardManagerClass']` will always be empty because it gets `unset()` a few lines prior as it is not part of `$nonRewrittenKeys` array.

So there is no way to provide a custom shard manager class via configs as it just gets unset.

Is this intentional or is it a bug?

### PR DESCRIPTION
**Pull request by cyberemissary**
Fixed a bug that does not allow to override the shard_manager_class

Closes #1314

### CODE REVIEW

File: Tests/DependencyInjection/DoctrineExtensionTest.php [hunk 1]
```diff
@@ -995,6 +1002,13 @@ public function testShardManager(): void
 
         $this->assertTrue($container->hasDefinition('doctrine.dbal.foo_shard_manager'));
         $this->assertFalse($container->hasDefinition('doctrine.dbal.bar_shard_manager'));
+        $this->assertTrue($container->hasDefinition('doctrine.dbal.baz_shard_manager'));
+
+        $fooManagerDef = $container->getDefinition('doctrine.dbal.foo_shard_manager');
+        $bazManagerDef = $container->getDefinition('doctrine.dbal.baz_shard_manager');
+
+        $this->assertEquals('Doctrine\\DBAL\\Sharding\\PoolingShardManager', $fooManagerDef->getClass());
```
Comment (ostrolucky) [inline Tests/DependencyInjection/DoctrineExtensionTest.php]: PoolingShardManager::class. Not sure why CS didn't complain.

**Review by ostrolucky — CHANGES_REQUESTED** [review]

File: Tests/DependencyInjection/DoctrineExtensionTest.php (hunk 1 above)
Comment (cyberemissary) [reply Tests/DependencyInjection/DoctrineExtensionTest.php]: Fixed

File: Tests/DependencyInjection/DoctrineExtensionTest.php [hunk 2]
```diff
@@ -975,8 +976,9 @@ public static function cacheConfigurationProvider(): array
 
     public function testShardManager(): void
     {
-        $container = $this->getContainer();
-        $extension = new DoctrineExtension();
+        $container    = $this->getContainer();
+        $extension    = new DoctrineExtension();
+        $managerClass = 'Fake\\ShardManager';
```
Comment (ostrolucky) [inline Tests/DependencyInjection/DoctrineExtensionTest.php]: Sorry, one more. I missed this in my last review

Comment (cyberemissary) [reply Tests/DependencyInjection/DoctrineExtensionTest.php]: Not sure what's here, did you want me to use a real class and reference with `::class` ?

Comment (ostrolucky) [reply Tests/DependencyInjection/DoctrineExtensionTest.php]: yes

**Review by ostrolucky — APPROVED** [review]

**Pull request merged**